src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -699,17 +699,37 @@ static ucs_status_t uct_rc_mlx5_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
-                                   const uct_device_addr_t *dev_addr,
-                                   const uct_iface_addr_t *iface_addr)
+int uct_rc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                      const uct_iface_is_reachable_params_t *params)
 {
-    uint8_t my_type = uct_rc_mlx5_iface_get_address_type(tl_iface);
+    const uct_iface_addr_t *iface_addr = params->iface_addr;
+    uct_ib_iface_t *iface              = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    uint8_t my_type                    = uct_rc_mlx5_iface_get_address_type(tl_iface);
 
     if ((iface_addr != NULL) && (my_type != *(uint8_t*)iface_addr)) {
+        if (params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING) {
+            snprintf(params->info_string, params->info_string_length,
+                     "IB iface %p: is unreachable due to address type mismatch, "
+                     "local=%d, remote=%d", iface, my_type, *(uint8_t*)iface_addr);
+        } else {
+            ucs_debug("IB iface %p: is unreachable due to address type mismatch, "
+                      "local=%d, remote=%d", iface, my_type, *(uint8_t*)iface_addr);
+        }
         return 0;
     }
 
-    return uct_ib_iface_is_reachable(tl_iface, dev_addr, iface_addr);
+    return uct_ib_iface_is_reachable_v2(tl_iface, params);
+}
+
+int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
+                                   const uct_device_addr_t *dev_addr,
+                                   const uct_iface_addr_t *iface_addr)
+{
+    return uct_iface_is_reachable_v2_wrapper(tl_iface,
```

Comment:
sorry for misleading, it seems like this wrapper is not needed all together, I thought that `uct_iface_is_reachable` is `inline` function, since that's not true, we can just move required wrapping directly inside of `uct_iface_is_reachable`:
```
int uct_iface_is_reachable(const uct_iface_h iface, const uct_device_addr_t *dev_addr,
                           const uct_iface_addr_t *iface_addr)
{
    const uct_iface_is_reachable_params_t params = {
        .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
                              UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
        .device_addr        = dev_addr,
        .iface_addr         = iface_addr
    };

    return uct_iface_is_reachable_v2(tl_iface, &params);
}
```
and initialize old `iface->ops.iface_is_reachable` pointer with some stub function like `ucs_empty_function_do_assert_void` or just NULL pointer

Reply:
Yes. 
But I think we still need a wrapper in this PR. It's because that not all components support v2(most of them is ucs_empty_function_return_unsupported) since 500 lines code change limitation. The v1 reachable api is still called in the wireup.c. I think once we support all components we can remove the wrapper and call uct_iface_is_reachable_v2 directly in uct_iface_is_reachable. 

Am I right? Thanks

Reply:
This is why I said in offline discussion that wrapping v1/v2 in first PR won't reduce it so much, I expected this could be last PR more likely, now it seems no way without extra step like this wrapper